pkg/kubelet/stats/cadvisor_stats_provider.go

```diff
@@ -192,6 +202,21 @@ func isPodManagedContainer(cinfo *cadvisorapiv2.ContainerInfo) bool {
 	return managed
 }
 
+// getcadvisorPodInfoFromPodUID returns a pod cgroup information by matching the podUID with its CgroupName identifier base name
+// In ContainerInfo data struct, the key for pod cgroup is one of the following format (<podUID> is the string of the pod UID)
+// "/.../kubepods/pod<podUID>"
+// "/.../kubepods/besteffort/pod<podUID>"
+// "/.../kubepods/burstable/pod<podUID>"
+// "/.../kubepods/guaranteed/pod<podUID>"
+func getcadvisorPodInfoFromPodUID(podUID types.UID, infos map[string]cadvisorapiv2.ContainerInfo) *cadvisorapiv2.ContainerInfo {
```

Comment:
did you test this w/ cgroup-driver=systemd

it doesn't appear that this will work as the name in cAdvisor will be different.  there is a conversion function that you can use based on the configured cgroup driver.